File: Caesar Cipher/Python/caesar_decrypt.py
# Code in Python to implement just the decryption part of Caesar Cipher algorithm

cipher_text = str(input("Enter cipher text: "))
key = int(input("Enter key: "))
plain_text = ""

for i in range(0, len(cipher_text)):
    char = cipher_text[i]

    # to add blank space
    if char == chr(32):
        plain_text += char
        continue

    # for upper-case characters
    elif (char.isupper()):
        plain_text += chr((ord(char) - key-65) % 26 + 65)

    # for lower-case characters
    elif (char.islower()):
        plain_text += chr((ord(char) - key-97) % 26 + 97)

    # special characters remains unchanged
    else:
        plain_text += char

print("Cipher Text:	", cipher_text)
print("Plain Text:	", plain_text)

File: Lab Work/Caesar Cipher/C/caesar_decrypt.c
#include <stdio.h>
#include <ctype.h> // for character classification and transformation

int main() {
    char cipher_text[100]; 
    int key;
    char plain_text[100];
    int i;

    printf("Enter cipher text: ");
    fgets(cipher_text, sizeof(cipher_text), stdin); // using fgets to read a full line

    printf("Enter key: ");
    scanf("%d", &key);

    // decrypting the cipher text
    for (i = 0; cipher_text[i] != '\0'; i++) {
        char char_current = cipher_text[i];

        // to add blank space
        if (char_current == ' ') {
            plain_text[i] = char_current;
            continue;
        }

        // for upper-case characters
        if (isupper(char_current)) {
            plain_text[i] = (char)((((char_current - 'A') - key + 26) % 26) + 'A'); 
        }

        // for lower-case characters
        else if (islower(char_current)) {
            plain_text[i] = (char)((((char_current - 'a') - key + 26) % 26) + 'a'); 
        } 

        // special characters remain unchanged
        else {
            plain_text[i] = char_current; 
        }
    }

    plain_text[i] = '\0'; // adding null-termination at the end of plain_text string

    // displaying the output
    printf("Cipher Text:  %s", cipher_text);
    printf("Plain Text:   %s", plain_text);

    return 0;
}
